aten/src/ATen/function_wrapper.py

```diff
@@ -967,14 +970,14 @@ def find_formal(formal_name, formals):
         option['const_mark'] = '' if option['inplace'] else ' const'
 
         is_method = 'method' in option['variants']
-        is_function = 'function' in option['variants']
+        is_namespace_function = 'function' in option['variants']
         is_factory_method = find_formal('TensorOptions', formals)
-        is_deprecated_factory_method = \
-            formals[0]['dynamic_type'] == 'Type' and option['return_type'] == 'Tensor' and option['deprecated']
+        is_deprecated_factory_method = len(formals) > 0 and \
+            formals[0]['dynamic_type'] == 'Type' and \
+            option['return_type'] == 'Tensor' and option['deprecated']
         needs_native_definition = not is_deprecated_factory_method
 
         has_dispatch = dispatch_tensor or dispatch_type
-        is_namespace_function = is_function and (has_dispatch or is_factory_method)
```

Comment:
I've verified that all other functions w/ `is_function = True` have `(has_dispatch or is_factory_method) = True` too. So I think relaxing the check here is fine.